src/Workspaces/CSharp/Portable/Formatting/Rules/SpacingFormattingRule.cs

```diff
@@ -174,22 +176,12 @@ public override AdjustSpacesOperation GetAdjustSpacesOperation(SyntaxToken previ
             {
                 return AdjustSpacesOperationZeroOrOne(optionSet, CSharpFormattingOptions.SpaceWithinSquareBrackets);
             }
-            else if (currentKind == SyntaxKind.CloseBracketToken && HasFormattableBracketParent(currentToken))
+            if (currentKind == SyntaxKind.CloseBracketToken && previousKind != SyntaxKind.OmittedArraySizeExpressionToken && HasFormattableBracketParent(currentToken))
             {
-                if (currentToken.Parent is ArrayRankSpecifierSyntax)
-                {
-                    var parent = currentToken.Parent as ArrayRankSpecifierSyntax;
-                    if ((parent.Sizes.Any() && parent.Sizes.First().Kind() != SyntaxKind.OmittedArraySizeExpression) || parent.Sizes.SeparatorCount > 0)
-                    {
-                        // int []: added spacing operation on open [
-                        // int[1], int[,]: need spacing operation
-                        return AdjustSpacesOperationZeroOrOne(optionSet, CSharpFormattingOptions.SpaceWithinSquareBrackets);
-                    }
-                }
-                else
-                {
-                    return AdjustSpacesOperationZeroOrOne(optionSet, CSharpFormattingOptions.SpaceWithinSquareBrackets);
-                }
+                // int []: added spacing operation on [
```

Comment:
Comments were kept from the original code, I had the same confusion when I first read that, will fix up.